eagle-xl/scripts/stellar_mass_evolution.py

```diff
@@ -0,0 +1,130 @@
+"""
+Plots the stellar mass density evolution.
+"""
+import matplotlib
+
+import unyt
+
+import matplotlib.pyplot as plt
+import numpy as np
+import sys
+import glob
+
+from swiftsimio import load
+
+from swiftsimio.statistics import SWIFTStatisticsFile
+
+from swiftpipeline.argumentparser import ScriptArgumentParser
+
+from velociraptor.observations import load_observation
+
+arguments = ScriptArgumentParser(
+    description="Creates a stellar mass density evolution plot, with added observational data."
+)
+
+snapshot_filenames = [
+    f"{directory}/{snapshot}"
+    for directory, snapshot in zip(arguments.directory_list, arguments.snapshot_list)
+]
+
+stats_filenames = [
+    f"{directory}/statistics.txt" for directory in arguments.directory_list
+]
+
+names = arguments.name_list
+output_path = arguments.output_directory
+
+plt.style.use(arguments.stylesheet_location)
+
+simulation_lines = []
+simulation_labels = []
+
+fig, ax = plt.subplots()
+
+ax.loglog()
+
+for snapshot_filename, stats_filename, name in zip(
+    snapshot_filenames, stats_filenames, names
+):
+    data = SWIFTStatisticsFile(stats_filename)
+
+    snapshot = load(snapshot_filename)
+    boxsize = snapshot.metadata.boxsize
+    box_volume = boxsize[0] * boxsize[1] * boxsize[2]
+
+    # a, Redshift, SFR
+    scale_factor = data.a
+    redshift = data.z
+    stellar_mass = data.star_mass
+    stellar_mass_density = stellar_mass / box_volume
+
+    # High z-order as we always want these to be on top of the observations
+    simulation_lines.append(
+        ax.plot(scale_factor, stellar_mass_density, zorder=10000)[0]
+    )
+    simulation_labels.append(name)
+
+# Observational data plotting
+
+observational_data = glob.glob(
+    f"{arguments.config.config_directory}/{arguments.config.observational_data_directory}/data/StellarMassDensity/*.hdf5"
+)
+
+observation_lines = []
+observation_labels = []
+
+for index, observation in enumerate(observational_data):
+    obs = load_observation(observation)
+    observation_lines.append(
```

Comment:
You should use `obs.plot_on_axes()`

Reply:
That doe leave out the label and does only work for one dataset.